pkg/odo/cli/component/common_push.go

```diff
@@ -27,7 +28,7 @@ type CommonPushOptions struct {
 	sourceType       config.SrcType
 	sourcePath       string
 	componentContext string
-	localConfigInfo  *config.LocalConfigInfo
+	LocalConfigInfo  *config.LocalConfigInfo
```

Comment:
Why do you need to expose `LocalConfigInfo`. You shouldn't need to do anything with this outside of this package

Reply:
I was considering writing functions for a couple of things that might be needed, let me see if I can do that

Reply:
Ok guess it might be harder as localconfiginfo is required for  a bunch of things more refactor will be needed to allow that

Reply:
It is exposed for common push. There might be a better way to do this but will require a lot more refactor
@kadel